lib/ts/recipe/thirdparty/types.ts

```diff
@@ -102,10 +101,14 @@ export type RecipeInterface = {
               status: "NO_EMAIL_GIVEN_BY_PROVIDER";
               fetchResponse: Response;
           }
-        | {
-              status: "FIELD_ERROR";
-              error: string;
-              fetchResponse: Response;
-          }
     >;
+
+    generateStateToSendToOAuthProvider: (input: { userContext: any; config: NormalisedInputType }) => string;
+
+    verifyStateFromOAuthProvider: (input: {
```

Comment:
Rename to `verifyAndGetStateOrThrowError` - this will need to be:
```
verifyAndGetStateOrThrowError({
   stateFromProvider: string | undefined;
   stateFromStorage: StateObject | undefined;
   config: NormalisedInputType;
   userContext: any;
}) => Promise<StateObject>
```